clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -189,6 +194,13 @@ public void run(long now) {
                                                                          result.readyNodes,
                                                                          this.maxRequestSize,
                                                                          now);
+        if (sendInOrder) {
+            // Mute all the partitions drained
+            for (Map.Entry<Integer, List<RecordBatch>> entry : batches.entrySet()) {
```

Comment:
Minor: doesn't look like you're using the key, maybe you could use `batches.values()`?